Refactor code via clippy

topecongiro opened the pull request:
This PR applies refactoring from rust-clippy.

sinkuu commented on src/expr.rs:
```diff
@@ -315,12 +305,12 @@ pub fn format_expr(
-    prefix: &str,
-    infix: &str,
-    suffix: &str,
+    pp: &PairParts,
```
I suppose you are trying to suppress needless_take_by_value lint with this useless reference... Consider adding `#[derive(Copy, Clone)]` to `PairParts` definition, and change the argument type to just `PairParts`. That lint ignores `Copy` types.

nrc commented on src/modules.rs:
```diff
@@ -76,6 +76,6 @@ fn module_file(
     match parser::Parser::default_submod_path(id, dir_path, codemap).result {
         Ok(parser::ModulePathSuccess { path, .. }) => path,
-        Err(_) => panic!("Couldn't find module {}", id),
+        Err(_) => unreachable!("Couldn't find module {}", id),
```
This probably shouldn't be `unreachable` - we're not saying its impossible to get here, we're just not handling the case where we do

nrc commented on src/expr.rs:
```diff
@@ -315,12 +305,12 @@ pub fn format_expr(
         })
 }
 
+pub struct PairParts<'a>(pub &'a str, pub &'a str, pub &'a str);
```
Could you make this a proper struct rather than tuple struct so that we have some 'documentation' of the fields please? Then we can avoid the destructuring at the top of the function too.

nrc commented on src/items.rs:
```diff
@@ -1434,17 +1434,46 @@ pub fn rewrite_struct_field(
+    expr_opt: Option<&'a ptr::P<ast::Expr>>,
+}
+
+impl<'a> StaticParts<'a> {
```
Could you use `derive(new)` rather than spell out this impl?

nrc commented on src/items.rs:
```diff
@@ -1434,17 +1434,46 @@ pub fn rewrite_struct_field(
+        ty,
+        mutability,
+        expr_opt,
+    } = *static_parts;
```
Could you use filed access rather than this destructuring?

topecongiro commented:
@sinkuu @nrc Thank you for the reviews! Rebased & updated. Added commits start from https://github.com/rust-lang-nursery/rustfmt/pull/2121/commits/1cd26db8c47af562d0e1bc78057c875c5e6f4e91.

nrc commented:
THanks!
